### PR TITLE
feat(webpack-cli): add no-mode flag

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**

feature - add `no-mode` flag

**Did you add tests for your changes?**
Work in progress.

**If relevant, did you update the documentation?**
Yes

**Summary**

Refers #1240 

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/groups/ZeroConfigGroup.js [hunk 1]
```diff
@@ -3,6 +3,13 @@ const { logger } = require('@webpack-cli/logger');
 
 const PRODUCTION = 'production';
 const DEVELOPMENT = 'development';
+const NONE = 'none';
+let isNoMode = false;
+process.argv.forEach((value) => {
```
Comment (ematipico) [inline packages/webpack-cli/lib/groups/ZeroConfigGroup.js]: That's not how you should retrieve the information.
Inside the constructor of this file, you have `options` and after the `super` call you will have access to `this.args` too.

Please retrieve the information from `options` and `this.args`

File: packages/webpack-cli/__tests__/ZeroConfigGroup.test.js [hunk 2]
```diff
@@ -42,4 +42,15 @@ describe('GroupHelper', function() {
...
+    it('should handle the mode option [none]', () => {
+        const group = new ZeroConfigGroup([
+            {
+                mode: 'none',
+            },
+        ]);
+
+        const result = group.run();
+        expect(result.options.mode).toEqual('development');
+    });
```
Comment (ematipico) [inline packages/webpack-cli/__tests__/ZeroConfigGroup.test.js]: The assertion is not correct. It should be `"none"`

File: packages/webpack-cli/lib/groups/ZeroConfigGroup.js [hunk 3]
```diff
@@ -27,13 +34,27 @@ class ZeroConfigGroup extends GroupHelper {
                 );
                 return this.args.mode;
             }
+            if (isNoMode && this.args.mode) {
+                logger.warn(
+                    'You Provided both mode and no-mode arguments. You Should Provide just one. "mode" will be used.'
+                )
+            }
+            if (isNoMode && (this.args.dev || this.args.prod)) {
+                logger.warn(
```
Comment (ematipico) [inline packages/webpack-cli/lib/groups/ZeroConfigGroup.js]: There's already the same message in this file. Please reuse it instead of copying

File: packages/webpack-cli/__tests__/ZeroConfigGroup.test.js (hunk 2 above)
Comment (snitin315) [reply packages/webpack-cli/__tests__/ZeroConfigGroup.test.js]: but when we pass `--mode=none` , it outputs development build .

Comment (snitin315) [reply packages/webpack-cli/__tests__/ZeroConfigGroup.test.js]: Also this test is failling if I replace it with `"none"`

Comment (ematipico) [reply packages/webpack-cli/__tests__/ZeroConfigGroup.test.js]: Which means it's a bug and code should be reviewed. You should rely on your testing suite, not on the code.
That's the point the test. (TDD)

Comment (snitin315) [reply packages/webpack-cli/__tests__/ZeroConfigGroup.test.js]: Screenshot from 2020-02-28 15-28-02

Comment (rishabh3112) [reply packages/webpack-cli/__tests__/ZeroConfigGroup.test.js]: Please keep the code for test the same and change the code which you are testing to make it pass the tests.

Comment (snitin315) [reply packages/webpack-cli/__tests__/ZeroConfigGroup.test.js]: @ematipico Yes there is a bug `--mode` is accepting any value and outputs development build. I can pass `--mode=abcd` , and it produces the build.

Screenshot from 2020-02-28 15-33-34

It should accept only `production , development or none`

**Comment by snitin315** [comment]
@ematipico @rishabh3112 , Yes there is a bug `--mode` is accepting any value and outputs development build. I can pass `--mode=abcd` , and it produces the build.

Screenshot from 2020-02-28 15-33-34

It should accept only `production , development or none`

Shoul I open a seperate issue for this? can you please verify?

**Comment by ematipico** [comment]
Argument validation will be part of another story. Don't worry about it.
Just focus on your feature for now.

**Comment by ematipico** [comment]
Could you rebase your branch please?

**Comment by snitin315** [comment]
@ematipico already done :+1:

File: test/help/__snapshots__/help-single-arg.test.js.snap [hunk 4]
```diff
@@ -25,28 +25,29 @@ Available Commands
...
+  -h, --hot                Enables Hot Module Replacement                       
+  -s, --sourcemap string   Determine source maps to use                         
+  --prefetch string        Prefetch this request                                
+  -j, --json               Prints result as JSON                                
+  --standard               Prints standard output                               
+  -d, --dev                Run development build                                
+  -p, --prod               Run production build                                 
+  --mode string            Defines the mode to pass to webpack                  
+  --no-mode                Sets mode="none" which disables any default behavior 
+  --version                Get current version
```
Comment (snitin315) [inline test/help/__snapshots__/help-single-arg.test.js.snap]: @ematipico isn't this the snapshot test?

**Comment by snitin315** [comment]
@ematipico can you help me fix the CI , dont know why scss test is failing.

**Comment by ematipico** [comment]
Please add a `it.skip` to `json.test.js`. That file is the culprit
